File: frontend/src/types/task.ts
export type TaskPriority = "low" | "medium" | "high";

export interface TaskComment {
  id: string;
  task_id: string;
  author_name: string;
  body: string;
  created_at: string;
}

export interface Task {
  id: string;
  title: string;
  description: string | null;
  section_id: string | null;
  priority: TaskPriority;
  category: string | null;
  estimated_mins: number | null;
  deadline: string | null;
  scheduled_for: string | null;
  done: boolean;
  created_at: string;
  updated_at: string;
  comments: TaskComment[];
}

export interface TaskCreate {
  title: string;
  description?: string;
  section_id?: string | null;
  priority?: TaskPriority;
  category?: string;
  estimated_mins?: number;
  deadline?: string;
  scheduled_for?: string;
}

export interface TaskUpdate {
  title?: string;
  description?: string;
  section_id?: string | null;
  priority?: TaskPriority;
  category?: string;
  estimated_mins?: number;
  deadline?: string;
  scheduled_for?: string;
  done?: boolean;
}

export interface TaskCounts {
  today: number;
  week: number;
}

File: backend/app/schemas/task.py
from __future__ import annotations
import uuid
from datetime import datetime, date
from typing import Optional, List
from pydantic import BaseModel, ConfigDict


class TaskCommentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    task_id: uuid.UUID
    author_name: str
    body: str
    created_at: datetime


class TaskCommentCreate(BaseModel):
    author_name: str = "You"
    body: str


class TaskBase(BaseModel):
    title: str
    description: Optional[str] = None
    status: str = "pending"
    priority: str = "medium"
    category: Optional[str] = None
    estimated_mins: Optional[int] = None
    deadline: Optional[datetime] = None
    scheduled_for: Optional[date] = None


class TaskCreate(TaskBase):
    pass


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    priority: Optional[str] = None
    category: Optional[str] = None
    estimated_mins: Optional[int] = None
    deadline: Optional[datetime] = None
    scheduled_for: Optional[date] = None
    done: Optional[bool] = None


class TaskOut(TaskBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    done: bool
    created_at: datetime
    updated_at: datetime
    comments: List[TaskCommentOut] = []


class TaskCounts(BaseModel):
    today: int = 0
    week: int = 0
    ongoing: int = 0
    pending: int = 0
    freetime: int = 0
